src/RazorLight/Compilation/RoslynCompilationService.cs

```diff
@@ -136,9 +139,25 @@ public Assembly CompileAndEmit(IGeneratedRazorTemplate razorTemplate)
 				}
 
 				assemblyStream.Seek(0, SeekOrigin.Begin);
-				pdbStream.Seek(0, SeekOrigin.Begin);
-
-				var assembly = Assembly.Load(assemblyStream.ToArray(), pdbStream.ToArray());
+				pdbStream.Seek(0, SeekOrigin.Begin);
+
+				var rawAssembly = assemblyStream.ToArray();
+				var rawSymbolStore = pdbStream.ToArray();
+				if (precreateAssemblyCallbacks != null)
+				{
+					foreach (var callback in precreateAssemblyCallbacks)
+					{
+						try
+						{
+							callback(razorTemplate, rawAssembly, rawSymbolStore);
+						}
+						catch (Exception)

```

Comment:
This is bad.

Reply:
In my defense I followed the exact same scheme as used for the already existing `prerenderCallbacks` - https://github.com/toddams/RazorLight/blob/22381db30ca3c61b4b1e4004e1536407dd138e49/src/RazorLight/TemplateRenderer.cs#L223-L239

I figured if that is good enough for them, why should it be different for these? However, I am will gladly change it, although I do not see any regression as opposed to the prerender callbacks.

Reply:
It's a fair point, but very few people actually utilize those callbacks because they're not easily discoverable. I've found most people follow a really old tutorial from medium.com for how to use RazorLight that is woefully out of date and contains some rather questionable decisions, that resulted in me spending way over a month figuring out why people were reporting bugs just "standing up RazorLight" that I couldn't fathom happening until I found that online tutorial's faulty guide.